lua/entities/gmod_wire_adv_microphone.lua

```diff
@@ -0,0 +1,206 @@
+AddCSLuaFile()
+
+local MIN_VOLUME = 0.02
+local MAX_DIST_GAIN = 1000
+
+local PLAYER_VOICE_MAXDIST_SQR = 250*250
+
+ENT.Type = "anim"
+ENT.Base = "base_wire_entity"
+ENT.Author = "stpM64"
+ENT.PrintName = "Wire Advanced Microphone"
+ENT.Purpose = "Listens to sounds, soundscapes and player voices"
+-- Named 'advanced' because 'gmod_wire_microphone' exists in Wire Extras
+ENT.WireDebugName = "Advanced Microphone"
+
+-- Note: we listen both serverside and clientside,
+-- because some sounds are played clientside only
+
+-- array(Entity(gmod_wire_adv_microphone))
+-- Array instead of lookup table because sounds are emitted more often than microphones switched on or off,
```

Comment:
Is there any notable difference in speed between ipairs and pairs in this case?

Cause I'd really prefer a lookup table over using `table.RemoveByValue` and `table.insert`. Even if I know there won't be more than maybe ten microphones placed in a server at once. At worst you could maintain both a lookup table and an array for the best of both worlds.

I know ipairs can jit but that's just on the x86_64 branch. Although you could use a manual for loop, but you aren't doing that.

Reply:
I can only agree on this. And I just don't get it why it such a problem to just change. It is not like you are going to add a extra 1000 line of code.